examples/step-12/step-12.cc

```diff
@@ -470,8 +471,17 @@ namespace Step12
   template <int dim>
   void AdvectionProblem<dim>::solve()
   {
-    SolverControl                    solver_control(1000, 1e-12);
-    SolverRichardson<Vector<double>> solver(solver_control);
+    SolverControl solver_control(1000, 1e-12);
+    // We create an additional data object for the GMRES solver to increase the
+    // maximum number of basis vectors of the Krylov subspace. When this number
+    // is reached the GMRES algorithm is restarted using the solution of the
+    // previous iteration as the starting approximation. The choice of the
+    // number of basis vectors is a trade- off between memory consumption and
+    // convergence speed, since a longer basis means minimization over a larger
+    // space.
```

Comment:
Can you move this block of documentation right after the text above the function that discusses the use of GMRES? The function will be easier to read in the documentation if more of the code remains together.